product_docs/docs/bdr/4.0/upgrades.mdx

```diff
@@ -79,6 +79,25 @@ in case of this example), so any new features provided by the newer version
 which require changes in the protocol will be disabled. Once all nodes are
 upgraded to the same version, the new features are automatically enabled.
 
+Similarly, when a cluster with WAL decoder enabled nodes is going through
+rolling upgrade, WAL decoder on higher version of BDR will produce LCRs with
+higher pglogical version and WAL decoder on lower version of BDR will produce
+LCRs with lower pglogical version. As a result, WAL senders on higher version
+of BDR node/s are not expected to use LCRs due to mismatch in protocol version
+while on lower version of BDR node/s, WAL senders may continue to use LCRs.
+This is because pglogical receiver on higher version BDR node/s will request
+higher protocol version range but the corrosponding WAL sender on lower version
+of BDR node/s will use pglogical version available on that node. That usually
+turns out to be the same pglogical version used to produce LCRs. Hence, on
+lower version of BDR node/s, WAL senders will use LCRs. On the other side, the
+pglogical receiver on lower version of BDR node/s will request lower protocol
+version range. The corresponding WAL sender on higher version of BDR node/s
+will use a lower pglogical version available on that node. This usually turns
+out to bea pglogical version lesser than the used to produce LCRs. Hence, on
+higher version of BDR node/s, WAL senders we will not use LCRs. In future
+version of BDR, WAL senders on higher version of BDR node/s may be able to
+convert the LCRs from one pglogical version to another.
+
```

Comment:
I think we need a shorter version like

Similarly, when a cluster with WAL decoder enabled nodes is going through
rolling upgrade, WAL decoder on higher version of BDR node will produce LCRs with
higher pglogical version and WAL decoder on lower version of BDR node will produce
LCRs with lower pglogical version. As a result, WAL senders on higher version
of BDR node/s are not expected to use LCRs due to mismatch in protocol version
while on lower version of BDR node/s, WAL senders may continue to use LCRs.
Once all the BDR nodes are on the same BDR version, WAL senders will use LCRs.

Reply:
Ok, updated PR with shorter version.